vllm/v1/engine/output_processor.py

```diff
@@ -205,13 +216,45 @@ def make_request_output(
             # Only the final output is required in FINAL_ONLY mode.
             return None
 
+        # Stream Interval buffering: only apply for DELTA mode and stream_interval > 1
+        is_delta_streaming = self.output_kind == RequestOutputKind.DELTA
+        if is_delta_streaming and self.stream_interval > 1:
+            # Track total tokens generated
+            self.total_num_output_tokens += len(new_token_ids)
+
+            # should send output when it is the first token or reach the stream interval
+            should_send_output = (
+                self.sent_tokens_offset == 0
+                or self.total_num_output_tokens - self.sent_tokens_offset
+                >= self.stream_interval
+            )
+
+            # Do NOT send output if not finished and should not send output
+            if not finished and not should_send_output:
+                return None
+
+            # Send tokens from the offset
+            assert self.detokenizer is not None
+            tokens_to_send = self.detokenizer.output_token_ids[
+                self.sent_tokens_offset :
+            ]
+            self.sent_tokens_offset = len(self.detokenizer.output_token_ids)
+        else:
+            if self.stream_interval > 1:
+                logger.warning_once(
+                    "stream_interval > 1 is only supported for DELTA mode. "
+                    "Ignoring the stream_interval setting."
+                )
+            # Send all the new tokens
+            tokens_to_send = new_token_ids
```

Comment:
Actually it should be trivial to support `CUMULATIVE` in the same way here (i.e. for the two `not final_only` cases), we just don't need to do the `tokens_to_send` slice in that case because that arg is ignored in  `_new_completion_output`.